### PR TITLE
[REM3-333] At RemoteReadListener, handle outboud channel closed when …

### PR DESCRIPTION
**Pull request by fl4via**
…reading a SERVICE_ERROR
Jira: https://issues.jboss.org/browse/REM3-333

### CODE REVIEW

File: src/main/java/org/jboss/remoting3/remote/RemoteReadListener.java [hunk 1]
```diff
@@ -173,7 +173,7 @@ public void handleEvent(final ConduitStreamSourceChannel channel) {
 
                             final RegisteredService registeredService = handler.getConnectionContext().getRegisteredService(serviceType);
                             if (registeredService == null) {
-                                refuseService(channelId, "Unknown service name");
+                                refuseService(channelId, "Unknown service name " + serviceType);
```
Comment (fl4via) [inline src/main/java/org/jboss/remoting3/remote/RemoteReadListener.java]: @dmlloyd if you're okay with adding service type at htis message, I'm going to create a Jira for it and submit this change

File: src/main/java/org/jboss/remoting3/remote/RemoteReadListener.java [hunk 2]
```diff
@@ -419,6 +419,7 @@ public void handleEvent(final ConduitStreamSourceChannel channel) {
                         case Protocol.SERVICE_ERROR: {
                             log.trace("Received service error");
                             int channelId = buffer.getInt() ^ 0x80000000;
+                            handler.handleOutboundChannelClosed(); // TODO how do I know it is outbound and not inbound?
```
Comment (fl4via) [inline src/main/java/org/jboss/remoting3/remote/RemoteReadListener.java]: @dmlloyd is this correct? I looked at the code, and it looked like I could make this assumption... or there is no way of knowing if it is an outbound or inbound channel that generated the message?

**Comment by fl4via** [comment]
Do no merge yet, I need feedback on the coments in the commit first (then I'll dismember it if appropriate and remove the comment)

File: src/main/java/org/jboss/remoting3/remote/RemoteReadListener.java (hunk 1 above)
Comment (dmlloyd) [reply src/main/java/org/jboss/remoting3/remote/RemoteReadListener.java]: Yeah this is fine

File: src/main/java/org/jboss/remoting3/remote/RemoteReadListener.java (hunk 2 above)
Comment (dmlloyd) [reply src/main/java/org/jboss/remoting3/remote/RemoteReadListener.java]: Yeah IIRC this can only happen on outbound channel open requests.

**Comment by fl4via** [comment]
I split this code into REM3-333 and REM3-334 and opened new PRs, one for each

**Pull request closed without merging**